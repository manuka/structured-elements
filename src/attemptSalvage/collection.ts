import type { StructuredElements } from "@"

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const attemptSalvageCollection: StructuredElements.Functions.AttemptSalvage<
  `collection`
> = <Element>({
  failures,
  subject,
  validElements,
}: {
  failures: StructuredElements.Failure[]
  subject: unknown
  validElements?: StructuredElements.Collection<Element>
}): StructuredElements.Collection<Element> | undefined => {
  if (failures.length === 0) {
    return subject as StructuredElements.Collection<Element>
  }

  if (validElements) {
    return validElements
  }

  return {}
}
