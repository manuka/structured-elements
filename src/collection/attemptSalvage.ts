import type { Validation } from "@/lib/validation"

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const attemptCollectionSalvage: Validation.Functions.AttemptSalvage<
  `collection`
> = <Element>({
  failures,
  subject,
  validElements,
}: {
  failures: Validation.Failure[]
  subject: unknown
  validElements?: Validation.Collection<Element>
}): Validation.Collection<Element> | undefined => {
  if (failures.length === 0) {
    return subject as Validation.Collection<Element>
  }

  if (validElements) {
    return validElements
  }

  return {}
}
