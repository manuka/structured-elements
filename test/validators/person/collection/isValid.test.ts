import { setupValidatorPersonCollectionTest } from "&/validators/person/collection/common"
import { testNestedScenarios } from "test-nested-scenarios"

describe(`validator('Person', 'collection').isValid(\n  {`, () => {
  const setup = setupValidatorPersonCollectionTest({
    expectation: ({ subject, validator }) => {
      it(`},\n  'Person'\n)`, () => {
        expect(validator.isValid(subject, `Person`)).toMatchSnapshot()
      })
    },
  })
  testNestedScenarios(setup)
})
