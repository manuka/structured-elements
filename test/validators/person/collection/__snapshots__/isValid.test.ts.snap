// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Invalid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Invalid), person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Invalid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Valid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Valid), person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: (Valid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: undefined, person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: undefined, person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Invalid), person1: undefined, person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Invalid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Invalid), person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Invalid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Valid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Valid), person2: (Valid), },
  'Person'
) 1`] = `true`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: (Valid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: undefined, person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: undefined, person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: (Valid), person1: undefined, person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Invalid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Invalid), person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Invalid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Valid), person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Valid), person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: (Valid), person2: undefined, },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: undefined, person2: (Invalid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: undefined, person2: (Valid), },
  'Person'
) 1`] = `false`;

exports[`validator('Person', 'collection').isValid(
  { person0: undefined, person1: undefined, person2: undefined, },
  'Person'
) 1`] = `false`;
